packages/plugins/plugin-advertising-id/src/AdvertisingIdPlugin.ts

```diff
@@ -4,25 +4,29 @@ import {
   PluginType,
   SegmentClient,
 } from '@segment/analytics-react-native';
+import { warnMissingNativeModule } from '@segment/analytics-react-native/src/util';
 import { NativeModules } from 'react-native';
 
 export class AdvertisingIdPlugin extends Plugin {
   type = PluginType.enrichment;
 
   configure(analytics: SegmentClient): void {
     this.analytics = analytics;
-
-    NativeModules.AnalyticsReactNativePluginAdvertisingId.getAdvertisingId().then(
-      (id: string) => {
-        if (id === null) {
-          analytics.track(
-            'LimitAdTrackingEnabled (Google Play Services) is enabled'
-          );
-        } else {
-          this.setContext(id);
+    if (NativeModules.AnalyticsReactNativePluginAdvertisingId) {
+      NativeModules.AnalyticsReactNativePluginAdvertisingId.getAdvertisingId().then(
+        (id: string) => {
+          if (id === null) {
+            analytics.track(
+              'LimitAdTrackingEnabled (Google Play Services) is enabled'
+            );
+          } else {
+            this.setContext(id);
+          }
         }
-      }
-    );
+      );
+    } else {
+      warnMissingNativeModule();
+    }
```

Comment:
nit: how about we have a function in `utils` for `getAnalyticsNativeModule` or something like that that either returns the module or `undefined` and triggers the `warnMissingNativeModule`

That way you can simplify the rest of the calls as:

```ts
getAnalyticsNativeModule()?.getAdvertisingId().then(
```

and 

```ts
getAnalyticsNativeModule()?.getContextInfo() ?? DefaultContext;
```